api/data_source.go

```diff
@@ -82,15 +109,16 @@ type DataSourceCreate struct {
 	DatabaseID int `jsonapi:"attr,databaseId"`
 
 	// Domain specific fields
-	Name         string         `jsonapi:"attr,name"`
-	Type         DataSourceType `jsonapi:"attr,type"`
-	Username     string         `jsonapi:"attr,username"`
-	Password     string         `jsonapi:"attr,password"`
-	SslCa        string         `jsonapi:"attr,sslCa"`
-	SslCert      string         `jsonapi:"attr,sslCert"`
-	SslKey       string         `jsonapi:"attr,sslKey"`
-	HostOverride string         `jsonapi:"attr,hostOverride"`
-	PortOverride string         `jsonapi:"attr,portOverride"`
+	Name         string            `jsonapi:"attr,name"`
+	Type         DataSourceType    `jsonapi:"attr,type"`
+	Username     string            `jsonapi:"attr,username"`
+	Password     string            `jsonapi:"attr,password"`
+	SslCa        string            `jsonapi:"attr,sslCa"`
+	SslCert      string            `jsonapi:"attr,sslCert"`
+	SslKey       string            `jsonapi:"attr,sslKey"`
+	HostOverride string            `jsonapi:"attr,hostOverride"`
+	PortOverride string            `jsonapi:"attr,portOverride"`
+	Options      DataSourceOptions `jsonapi:"attr,options"`
```

Comment:
Did u test the create/update from the frontend to backend? Far as I know seems the JSONAPI doesn't support the nested struct

Reply:
+1

Reply:
jsonapi supports marshal struct types: https://go.dev/play/p/bGkP5CkCREf
but not support **nested** struct: https://go.dev/play/p/SVoy2cOpFk_Z

Reply:
The JSON keys in the output are determined by the field names in the struct, not the jsonapi tags. If you want the JSON keys to match the jsonapi tags exactly, you can use the json struct tag instead of the jsonapi tag. So I define DataSourceOptions Struct as: 
```go
type DataSourceOptions struct {
	// SRV is used for MongoDB only.
	SRV bool `json:"srv" jsonapi:"attr,srv"`
}
```
NOTE: `json:"srv" jsonapi:"attr,srv"`
Here is the another example may help you:https://go.dev/play/p/6IeXzE2jLnr